### PR TITLE
Add support for passing in a port other than 443

### PR DESCRIPTION
**Pull request by agrare**

### CODE REVIEW

**Comment by kbrock** [comment]
Does feel like an optional parameter. But requiring it is probably not the end of the world.

Where do you want to go with this? The spots where I see changes are (`grep MiqVim.new`):

- `manageiq/tools`
  - [ ] `list_evm_snapshots.rb`
  - [ ] `rm_evm_snapshots.rb`
- `manageiq-providers-vmware/app/models/manageiq/providers/vmware/infra_manager/`
  - [ ] `vim_connect_mixin.rb`
  - [ ] `scanning/job.rb`
  - [ ] `vm_or_template_shared/scanning.rb`
- `manageiq-smartstate/lib/`
  - [ ] `MiqVm/test/partitionAlignmentCheck.rb`
  - [ ] `MiqVm/test/remoteVm.rb`
  - [ ] `metadata/VmConfig/VmConfig.rb`

**Comment by agrare** [comment]
> Does feel like an optional parameter. But requiring it is probably not the end of the world.

Yeah I agree I didn't want to just stick it at the end of all those other optional positional arguments.

While we're breaking the interface maybe switching to kwargs would be cleaner

File: lib/VMwareWebService/MiqVim.rb [hunk 1]
```diff
@@ -23,18 +23,19 @@ class MiqVim < MiqVimInventory
...
   # @param monitor_updates [Bool] Should a thread be started to monitor updates (default: false)
-  # @param preLoad [Bool] Should the cache be built before returning the connection (default: false)
+  # @param pre_load [Bool] Should the cache be built before returning the connection (default: false)
   # @param debugUpdates [Bool] Should we print debug info for each update (default: false)
   # @param notifyMethod [Method] A optional method to call for each update (default: nil)
   # @param maxWait [Integer] How many seconds to wait before breaking out of WaitForUpdates (default: 60)
   # @param maxObjects [Integer] How many objects to return from each WaitForUpdates page (default: 250)
-  def initialize(server, username, password, cacheScope = nil, monitor_updates = nil, preLoad = nil, debugUpdates = false, notifyMethod = nil, maxWait = 60, maxObjects = 250)
-    super(server, username, password, cacheScope)
+  def initialize(server:, username:, password:, port: 443, cache_scope: nil, monitor_updates: nil, pre_load: nil, debugUpdates: false, notifyMethod: nil, maxWait: 60, maxObjects: 250)
```
Comment (chessbyte) [inline lib/VMwareWebService/MiqVim.rb:34]: kwargs are inconsistent with some as snake_case and others as camelCase.

Comment (agrare) [reply lib/VMwareWebService/MiqVim.rb:34]: Yeah while we're breaking the interface anyway I'm moving these over to all snake case, not done with all of them yet

**Review by Fryguy — APPROVED** [review]

**Pull request merged**